docs/content/preview/api/ysql/user-defined-subprograms-and-anon-blocks/language-plpgsql-subprograms/plpgsql-syntax-and-semantics/executable-section/basic-statements/doing-sql-from-plpgsql.md

```diff
@@ -0,0 +1,411 @@
+
+(You can reduce the size of the hit-list to about _300 thousand_ by searching for _PostgreSQL "format()" "SQL injection"_. )
+
+You must _always_ must the _format()_ built-in SQL function when you need to define the text of a dynamic SQL statement at run-time. The example above shows you how to do this.
+{{< /tip >}}
+
+### The "perform" statement
+
+_[See the syntax rule above](./#plpgsql-perform-stmt)_
+
+The _perform_ statement lets you execute a select statement that, as a top-level SQL statement, would produce one or many rows. But, critically, it lets you avoid using the _select into_ statement for one row or the _[query for loop]((../../compound-statements/loop-exit-continue/query-for-loop/))_ (or the _[infinite loop]((../../compound-statements/loop-exit-continue/query-for-loop/))_) for many rows. It's the moral equivalent of this:
+
+```output
+select ... into null;
+```
+
+analogous, in Unix, to sending output to _/Dev/Null_. Try this top-level SQL statement:
+
+```plpgsql
+with s(val) as (select g.val from generate_series(1, 11) as g(val))
+select s.val as k, s.val*10 as v from s;
+```
+
+This, of course, is the result:
+
+```output
+ k | v  
+---+----
+ 1 | 10
+ 2 | 20
+ ......
+ 8 | 80
+ 9 | 90
+```
+
+To use this as the argument for _perform_, you must encapsulate as a subquery in an outer _select_ statement, thus:
+
+```plpgsql\
+select k, v from (
+    with s(val) as (select g.val from generate_series(1, 11) as g(val))
+    select s.val as k, s.val*10 as v from s
+  ) as subqry;
+```
+
+You can now use this in a PL/pgSQL block statement simply by replacing the starting _select_ with _perform_. Do this:
+
+```plpgsql
+do $body$
+begin
+  perform k, v from (
+      with s(val) as (select g.val from generate_series(1, 11) as g(val))
+      select s.val as k, s.val*10 as v from s
+    ) as subqry;
+end;
+$body$;
+```
+
+It finishes silently without error.
+
+This _do_ statement immediately above shows the general case, where the _select_ statement whose results you want to throw away must use a _with clause_.
+
+ This need arises, for example, when you perform multi-table _insert_, _update_ or _delete_ by encapsulating these statements each with a pro-forma _returning_ clause, and in its own _common table expression (hereinafter _CTE_). Then you cause them to execute by selecting the union of these pro-forma results  from the various _CTEs_. But the results are of no interest. They were simply a necessary device.
+
+<!--- _to_do_ --->
+{{< note title="Coming soon" >}}
+A self-contained, working example will follow will follow.
+{{< /note >}}
+
+By for the most common use of _perform_ is when you need to use a SQL built-in, like _pg_sleep()_ that is morally a procedure but that, for implementation reasons, has to be a function. Try this:
```

Comment:
By far

Reply:
It's unlike you, Dwight, to have missed my careless doubling in "working example will follow will follow". Anyway, I fixed that.

Good catch on "By for". Fixed now.